### PR TITLE
fix(link): clear router-link/href collision & rm old mixin

### LINKED ISSUE #1015: 'href' attribute is not set in anchors when I use `:to` property in 'b-btn' component to set route by name
I use the `:to` property with my route name, but still renders an `<a>` element with `href="#"`.

Link still works; seems to handle a kind of `onclic` event. No problem. 

But I expected `href` to be set to the exact route.

**Router config:**
```js
const routes = [
  //...
  { path: '/auth/login', name: 'auth.login', component: require('./pages/auth/login.vue') },
  { path: '/auth/register', name: 'auth.register', component: require('./pages/auth/register.vue') },
  // ...
]

const router = new Router({
    routes,
    scrollBehavior,
    mode: 'history'
})
```

**I use:**
```html
<b-btn variant="primary" :to="{ name: 'auth.register' }">Sign up</b-btn>
```

**Renders to:**
```html
<a href="#" class="btn btn-primary" target="_self">Sign up</a>
```
> It adds an event listener for `click`.

**I expect:**
```html
<a href="/auth/login" class="btn btn-primary" target="_self">Sign up</a>
```

### Versions
```
yarn list v0.27.5
├─ bootstrap-vue@1.0.0-beta.7
├─ bootstrap@4.0.0-beta
├─ vue-router@2.7.0
└─ vue@2.4.2
```

### PR DESCRIPTION
**Pull request by alexsasharegan**
Fixes #1015

### CODE REVIEW

**Comment by alexsasharegan** [comment]
Waiting on @tmorehouse's fixes for `utils/dom` which are currently not resolving.

**Comment by tmorehouse** [comment]
Might want to merge in #940 as well, which checks for existence of $router, and fallsback to `href` and `<a>`

File: lib/components/link.js [hunk 1]
```diff
@@ -85,19 +85,29 @@ export const computed = {
...
 
 function computeHref({ disabled, href, to }, tag) {
+    // We've already checked the parent.$router in computeTag,
+    // so router-link means live router.
     // When deferring to Vue Router's router-link,
     // don't use the href attr at all.
-    if (tag === "router-link") return null;
+    // Must return undefined for router-link to populate href.
+    if (tag === "router-link") return void 0;
     if (disabled) return "#";
```
Comment (tmorehouse) [inline lib/components/link.js]: Do we need to set this to `#` now when disabled?

Comment (alexsasharegan) [reply lib/components/link.js]: I personally don't see the reason for this either. I think that ideally the link value wouldn't change while disabled.

File: docs/components/form-checkbox/README.md [hunk 2]
```diff
@@ -144,15 +143,14 @@ Note that when `v-model` is bound to multiple checkboxes (i.e an array ref), the
...
 When binding multiple checkboxes together, you should set the `name` prop to the same value for
 all checkboxes in the group, as well as wrap the group in a `<div>` (or other block element)
-which has the aria attribute `role="group"`. This will inform users of assitive technologies
+which has the aria attribute `role="group"`. This will inform users of assistive technologies
 that the checkboxes are related.
 
-When placing the group of checkboxes inside a `<b-form-groupt>`, set a unique `id` on the
-element with `role="group"` and set the `label-for` prop of the `<b-form-fieldet>` to
+When placing the group of checkboxes inside a `<b-form-group>`, set a unique `id` on the
+element with `role="group"` and set the `label-for` prop of the `<b-form-fieldset>` to
```
Comment (tmorehouse) [inline docs/components/form-checkbox/README.md]: `<b-form-fieldset>` should be `<b-form-group>`

File: lib/utils/dom.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ dom.isVisible = function(el) {
 
 // Determine if an element is disabled
 dom.isDisabled = function(el) {
-    return !dom.isElemetn(el) ||
+    return !dom.isElement(el) ||
```
Comment (tmorehouse) [inline lib/utils/dom.js]: 👍

**Comment by alexsasharegan** [comment]
should we merge this PR, or wait for a fix on that form-checkbox page?

**Comment by tmorehouse** [comment]
We could merge, and then create a separate fix for checkbox.

**Comment by nelson6e65** [comment]
Hooray! :tada: I'll wait for this in the the new beta release!

**Pull request merged**